src/Provider/LocationIQ/LocationIQ.php

```diff
@@ -108,6 +142,49 @@ public function reverseQuery(ReverseQuery $query): Collection
         return new AddressCollection([$this->xmlResultToArray($result, $addressParts)]);
     }
 
+    /**
+     * @param array $arrayResult
+     * @return Location
+     */
+    private function arrayResultToArray(array $arrayResult): Location
```

Comment:
Name of the function is confusing I think. I would name it something like `resultToLocation()`.